client/tokenized-express-checkout/blocks/components/express-checkout-button-preview.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useMemo, useRef } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { getExpressCheckoutButtonAppearance } from 'wcpay/express-checkout/utils';
+
+const ExpressCheckoutButtonPreview = ( {
+	expressPaymentMethod,
+	options,
+	buttonAttributes,
+} ) => {
+	const appearance = useMemo(
+		() => getExpressCheckoutButtonAppearance( buttonAttributes ),
+		[ buttonAttributes ]
+	);
+	const ref = useRef( null );
+	const renderGooglePayButtonPromise = useRef( null );
+
+	const theme = options.buttonTheme[ expressPaymentMethod ];
+	const borderRadius = appearance.variables.borderRadius;
+
+	useEffect( () => {
+		if (
+			ref.current &&
+			expressPaymentMethod === 'googlePay' &&
+			! renderGooglePayButtonPromise.current
+		) {
+			renderGooglePayButtonPromise.current = ( async () => {
+				const targetDocument = ref.current.ownerDocument;
+				const targetWindow = targetDocument.defaultView;
+				if ( ! targetWindow.google?.payments?.api?.PaymentsClient ) {
+					await new Promise( ( resolve ) => {
+						const script = document.createElement( 'script' );
+						script.src = 'https://pay.google.com/gp/p/js/pay.js';
+						script.onload = resolve;
+						targetDocument.head.appendChild( script );
+					} );
+				}
+
+				const googlePayClient = new targetWindow.google.payments.api.PaymentsClient(
+					{
+						environment: 'TEST',
+					}
+				);
+
+				const buttonColor = theme === 'black' ? 'black' : 'white'; // There is no 'outline' theme in Google Pay.
+
+				const button = googlePayClient.createButton( {
+					buttonType: 'plain',
+					buttonColor,
+					buttonRadius: parseFloat( borderRadius ),
+					buttonSizeMode: 'fill',
+					onClick: () => {},
+				} );
+				ref.current.appendChild( button );
+			} )();
+		}
+	}, [ ref, theme, expressPaymentMethod, borderRadius ] );
```

Comment:
```suggestion
	}, [ theme, expressPaymentMethod, borderRadius ] );
```

Return values from `useRef`s shouldn't be used as part of hook dependencies.